tests/components/google_photos/conftest.py

```diff
@@ -100,53 +109,73 @@ def mock_user_identifier() -> str | None:
     return USER_IDENTIFIER
 
 
+@pytest.fixture(name="api_http_status")
+def mock_api_http_status() -> http.HTTPStatus:
+    """Provide a json fixture file to load for list media item api responses."""
+    return http.HTTPStatus.OK
+
+
 @pytest.fixture(name="setup_api")
 def mock_setup_api(
-    fixture_name: str, user_identifier: str
+    aioclient_mock: AiohttpClientMocker,
+    fixture_name: str,
+    user_identifier: str,
+    api_http_status: http.HTTPStatus,
 ) -> Generator[Mock, None, None]:
     """Set up fake Google Photos API responses from fixtures."""
-    with patch("homeassistant.components.google_photos.api.build") as mock:
-        mock.return_value.userinfo.return_value.get.return_value.execute.return_value = {
+
+    aioclient_mock.get(
```

Comment:
We should patch the library instead of the aiothttp calls

Reply:
If that is now the requirement, i'm can do this. However, I am not convinced that it is always the right behavior. Recently I converted the flume tests to use `aioclient_mock` in https://github.com/home-assistant/core/pull/120851 because #120618 was not actually exercising a real client library.

I acknowledge there is definitely a down side for home assistant maintenance that the home assistant tests are taking on more responsibility than they absolutely need to. I found like ~5 bugs I made when implementing the client library by using these tests, though granted i could have also found those with manual testing (and I prefer aioclient_mock over manual testing).

Reply:
Tests in the HA project should test the HA code and tests in the library should test the library.

We want to keep logic in the library to make sure we don't have service or device specific code in the HA codebase, so I also think we should test those specifics in the library. If any of the outcomes have a change in the way the integration works it can become another test case. 

I personally do use data objects to store the mock data of the tests in my integrations because that's easier

Reply:
Ok, I'll make this change.

To be clear I am not saying home assistant test coverage replaces library coverage.

The gap is the area between the two things you mentioned. As a result of this there are no tests the library is actually called correctly. As a result we'll have more possible breaking on (1) dependency upgrades and (2) code cleanup and refactoring. This means more reliance more on manual and beta testing. (This makes sense why there are so many patches submitted during beta.).  While well intentioned, there are numerous super complex and brittle tests written in piles of mocks. I am also not making a case this is always the best approach.

I agre there are plenty of cases where it totally would be way too complex to exercise a complex protocol in a home assistant test case. I would not argue doing that either.

I would recommend reconsidering this to a more pragmatic view. Use of aioclient mocks are simple like in the two examples I am discussing. (The Google APIs this is replacing are extremely complex to mock and is where the guidance has intended consequences)

I'll make this change here and revert the flume PR.  (My impression there are many other integrations doing this because it's simple but I cant remember them all.)

Reply:
This has been updated to use mocks.

Here are examples of caveats of this approach for why i don't think it's great:
- Now home assistant has to depend on the internal details of the client library serialization details in order to continue to use the data objects from a fixture file.   In #121782 we talked about how home assistant shouldn't depend on details like this (though allow it when not being imported directly) however there isn't a simple replacement for https://github.com/home-assistant/core/blob/d68ee8dceae7acd2fbdc2be0724847ac0827dcde/tests/components/mealie/conftest.py#L56 so we can't break the dependency.  Another example where this approach was painful is the switch from pydantic to mashumaro (an internal client library detail that home assistant is exposed to when you take this approach) 
- It now has to mock out an async iterator because the iterator can only be constructed by the client library given it depends on a private object. Was pretty complex and took awhile to figure this out how to do this since its not super common.

Reply:
> I'll make this change here and revert the flume PR. 

After our discussion offline about this not being a requiremet i won't revert the flume PR, and will leave these test changes  in this PR for now.